tests/test-app/tests/utils/function/rendering-test.gts

```diff
@@ -226,4 +242,50 @@ module('Utils | trackedFunction | rendering', function (hooks) {
     await settled();
     assert.dom('out').containsText('12.206');
   });
+
+  test('failing case', async function (assert) {
+    class TestCase {
+      lengthPromise = trackedFunction(this, async () => {
+        return await Promise.resolve(3);
+      });
+
+      get length() {
+        return this.lengthPromise.value;
+      }
+
+      stringArray = trackedFunction(this, async () => {
+        if (!this.length) return [];
+
+        const stringArray = await Promise.resolve(
+          Array.from({ length: this.length }, () => 'item')
+        ); // ['item', 'item', 'item']
+
+        return stringArray;
+      });
+
+      get endResult() {
+        return this.stringArray.value?.join(',') ?? '';
+      }
+    }
+
+    class TestComponent extends Component {
+      @tracked testCase?: TestCase;
+
+      setTestCase = () => (this.testCase = new TestCase());
+
+      <template>
+        <div {{logText this.testCase.endResult}} />
+        <out>{{this.testCase.endResult}}</out>
+        <button type="button" {{on "click" this.setTestCase}}></button>
```

Comment:
The secret sauce to get the test to fail was using the value in a modifier before using it for something else.

This test passes when either;
* you comment out the div with the modifier, or
* the value is referenced before the modifier (eg adding `{{log this.testCase.endResult}}` to the line before the div

Reply:
> This test passes when either;

* or, my production builds seem to behave OK. I can't verify this here in the test though.